mypy/checkexpr.py

```diff
@@ -4007,12 +4027,19 @@ def is_valid_var_arg(self, typ: Type) -> bool:
 
     def is_valid_keyword_var_arg(self, typ: Type) -> bool:
         """Is a type valid as a **kwargs argument?"""
+        mapping_type = self.chk.named_generic_type(
+            'typing.Mapping', [self.named_type('builtins.str'), AnyType(TypeOfAny.special_form)])
+        typ = get_proper_type(typ)
+
         ret = (
-                is_subtype(typ, self.chk.named_generic_type('typing.Mapping',
-                    [self.named_type('builtins.str'), AnyType(TypeOfAny.special_form)])) or
-                is_subtype(typ, self.chk.named_generic_type('typing.Mapping',
-                    [UninhabitedType(), UninhabitedType()])) or
-                isinstance(typ, ParamSpecType)
+            is_subtype(typ, mapping_type) or
+            (isinstance(typ, Instance) and
+                is_subtype(typ, self.chk.named_type('typing.Mapping')) and
+                try_getting_str_literals_from_type(map_instance_to_supertype(
+                    typ, mapping_type.type).args[0]) is not None) or
+            is_subtype(typ, self.chk.named_generic_type('typing.Mapping',
+                                                        [UninhabitedType(), UninhabitedType()])) or
```

Comment:
Actually I don't know how this line works. Could somebody elaborate? @JukkaL @momohatt

Reply:
Sorry for being late.
As we discussed in #9629, this check is added to avoid false-positive errors when empty dictionaries are passed with double-stars (e.g., `**{}`). This condition is needed because the type of an empty dictionary is inferred to be `dict[<nothing>, <nothing>]`, which is not a subtype of `Mapping[str, Any]`.

Reply:
@momohatt Thanks!